contrib/storage-http/src/test/java/org/apache/drill/exec/store/http/TestHttpPlugin.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.http;
+
+import org.apache.drill.common.types.TypeProtos;
+import org.apache.drill.exec.physical.rowSet.RowSet;
+import org.apache.drill.exec.physical.rowSet.RowSetBuilder;
+import org.apache.drill.exec.record.metadata.SchemaBuilder;
+import org.apache.drill.exec.record.metadata.TupleMetadata;
+import org.apache.drill.exec.store.StoragePluginRegistry;
+import org.apache.drill.test.ClusterFixture;
+import org.apache.drill.test.ClusterTest;
+import org.apache.drill.test.QueryBuilder;
+import org.apache.drill.test.rowSet.RowSetComparison;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertEquals;
+
+@Ignore("requires remote http server")
```

Comment:
There is a way to run a simple HTTP server in your test. (We run the Drill HTTP server, for example.) I'd recommend adding that, then using it to return a variety of outputs:

* Error code
* Empty response
* Normal size response (that can be verified in test)
* Odd format (that requires, say, all text mode)
* Very large output (which will blow up the current load-to-string implementation)
* Query parameters
* Complex filters (including OR clauses which can be converted to multiple GETs)

For example, here is [the second hit](https://www.logicbig.com/tutorials/core-java-tutorial/http-server/http-server-basic.html) on a Java search for "java simple in-memory HTTP server".

Reply:
I completely redid the unit tests.  Since the code reuses the Drill JSONReader, the unit tests focus on the actual HTTP request.